src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumClient.kt

```diff
@@ -53,11 +54,11 @@ internal data class BroadcastStatus(val connection: Connection) : ElectrumClient
  *
  */
 internal sealed class ClientState {
-    abstract fun process(event: ClientEvent): Pair<ClientState, List<ElectrumClientAction>>
+    abstract fun process(event: ClientEvent, logger: Logger): Pair<ClientState, List<ElectrumClientAction>>
```

Comment:
This looks tedious but:
- it makes sense to use a logger context attached to the Machine and not the State of the FSM
- it prepares the use of [context receivers](https://kotlinlang.org/docs/whatsnew1620.html#prototype-of-context-receivers-for-kotlin-jvm)